posts/2022-03-17-graphql-22004-beta.adoc

```diff
@@ -0,0 +1,327 @@
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= MicroProfile GraphQL 2.0, Java 18 support, and distributed security caches on Open Liberty 22.0.0.4-beta
+Ryan Storey <https://github.com/ryan-storey>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+Open Liberty 22.0.0.4-beta offers GraphQL 2.0, which incorporates Jakarta EE 9.1 dependencies, and Java 18 support, which includes many new exciting features and changes. Also included in this release is distributed security cache support, which allows multiple Liberty servers to share caches via a JCache provider.
+
+The link:{url-about}[Open Liberty] 22.0.0.4-beta includes the following beta features and function (along with all GA features and function):
+
+* <<graphql, MicroProfile GraphQL 2.0>>
+* <<java18, Java 18 Support>>
+* <<security_caches, Distributed Security Caches>>
+
+[#graphql]
+== MicroProfile GraphQL 2.0
+
+MicroProfile GraphQL 2.0 incorporates Jakarta EE 9.1 dependencies.  This allows developers to continue using the same functionality provided by MP GraphQL 1.0 but with the updated Jakarta features such as link:{url-prefix}/docs/latest/reference/feature/cdi-3.0.html[CDI 3.0], link:{url-prefix}/docs/latest/reference/feature/restfulWS-3.0.html[restfulWS] 3.0, link:{url-prefix}/docs/latest/reference/feature/jsonb-2.0.html[JSON-B 2.0], etc. Functionally, version 2.0 behaves the same as 1.0 because it uses the same version of the underlying implementation (link:https://github.com/smallrye/smallrye-graphql[SmallRye GraphQL]).  If you'd like to learn more about MP GraphQL, see our previous link:{url-prefix}/blog/2020/06/10/microprofile-graphql-open-liberty.html[blog post about MP GraphQL 1.0] and our link:{url-prefix}/blog/2020/08/28/graphql-apis-open-liberty-20009.html[blog post on how to use the GraphiQL UI and third-party Client APIs].
+
+To enable the MicroProfile GraphQL 2.0 beta feature in your app, add it to your `server.xml`:
+
+[source, xml]
+----
+<featureManager>
+  <feature>mpGraphQL-2.0</feature>
+</featureManager>
+----
+
+[#java18]
+== Java 18 Support
+
+Java 18 is coming soon, and with it the following features and changes:
+
+* link:https://openjdk.java.net/jeps/400[UTF-8 by Default]
+* link:https://openjdk.java.net/jeps/408[Simple Web Server]
+* link:https://openjdk.java.net/jeps/413[Code Snippets in Java API Documentation]
+* link:https://openjdk.java.net/jeps/416[Reimplement Core Reflection with Method Handles]
+* link:https://openjdk.java.net/jeps/417[Vector API (Third Incubator)]
+* link:https://openjdk.java.net/jeps/418[Internet-Address Resolution SPI]
+* link:https://openjdk.java.net/jeps/419[Foreign Function & Memory API (Second Incubator)]
+* link:https://openjdk.java.net/jeps/420[Pattern Matching for switch (Second Preview)]
+* link:https://openjdk.java.net/jeps/421[Deprecate Finalization for Removal]
+
```

Comment:
[NOTE] 
====
Java Security support has been removed with Java 18+ in Liberty.  If "websphere.java.security" is set in `bootstrapping.properties` will produce the following error in the log:
```
CWWKE0955E: Java Security was requested in bootstrapping.properties while using JDK "18", however this option is no longer valid when using Java 18 and later.
```
====